pkg/manifest/mantaray/marshal.go

```diff
@@ -256,13 +256,14 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 			f := &fork{}
 
 			if len(data) < offset+nodeForkPreReferenceSize+refBytesSize {
-				err := fmt.Errorf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize))
-				return fmt.Errorf("%s on byte '%x': %w", err, []byte{b}, ErrInvalidManifest)
+				cause := fmt.Sprintf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize))
+				return fmt.Errorf("%s on byte '%x': %w", cause, []byte{b}, ErrInvalidManifest)
 			}
 
 			err := f.fromBytes(data[offset : offset+nodeForkPreReferenceSize+refBytesSize])
 			if err != nil {
-				return fmt.Errorf("%s on byte '%x': %w", err, []byte{b}, ErrInvalidManifest)
+				cause := err.Error()
+				return fmt.Errorf("%s on byte '%x': %w", cause, []byte{b}, ErrInvalidManifest)
```

Comment:
neat: would it make sense to inline `err.Error()` in `fmt.Errorf` arguments list (to make code a bit more compact)?
If you decide to this, please update similar pattern on other occurrences in PR.